### PR TITLE
1247890: KeyErrors are now caught when checking manager capabilities

### PR DESCRIPTION
**Pull request by cnsnyder**
This accounts for the fact that candlepin and katello return different info from a GET request to their /status resources. Candlepin provides managerCapabilities while Katello does not.

### CODE REVIEW

File: src/rhsm/connection.py [hunk 1]
```diff
@@ -775,9 +775,12 @@ def _load_manager_capabilities(self):
         """
         self.capabilities = {}
         self.capabilities = self.conn.request_get("/status/")
```
Comment (alikins) [inline src/rhsm/connection.py]: The initial set to {} seems unneeded.
Just return the results of get /status here (or just the part you need).

maybe
```
return status_results.get('managerCabalities', [])
```

File: src/rhsm/connection.py [hunk 2]
```diff
@@ -775,9 +775,12 @@ def _load_manager_capabilities(self):
...
-        self.capabilities = self.capabilities['managerCapabilities']
-        log.debug("Server has the following capabilities: %s",
-                  self.capabilities)
+        try:
+            self.capabilities = self.capabilities['managerCapabilities']
+            log.debug("Server has the following capabilities: %s",
+                    self.capabilities)
+        except KeyError as e:
+            log.debug("Unable to retrieve managerCapabilities: %s" % str(e))
```
Comment (alikins) [inline src/rhsm/connection.py]:
```
log.debug(""Unable to retrieve managerCapabilities: %s" , e)
```
That avoids evaluating e until the logger is prepared to emit the message (ie, if logs are set at INFO, str/repr of e doesn't need to happen and is skipped).

File: src/rhsm/connection.py [hunk 3]
```diff
@@ -858,6 +859,7 @@ def hypervisorCheckIn(self, owner, env, host_guest_mapping):
             query_params = urlencode({"owner": owner, "env": env})
             url = "/hypervisors?%s" % (query_params)
             res = self.conn.request_post(url, host_guest_mapping)
+        log.debug('lolol')
```
Comment (Ceiu) [inline src/rhsm/connection.py]: Extraneous debug output. :)

Comment (cnsnyder) [inline src/rhsm/connection.py]: :astonished: how did that get there?!

**Comment by dgoodwin** [comment]
**Build Log**
*last 10 lines*

 ```
[...truncated 173 lines...]
[htmlpublisher] Archiving at PROJECT level /home/jenkins/workspace/python-rhsm-pull-request-builder/test/unit/htmlcov to /jenkins-results/rhsm-jenkins/jobs/python-rhsm-pull-request-builder/htmlreports/Coverage_module_html_report
ERROR: Specified HTML directory '/home/jenkins/workspace/python-rhsm-pull-request-builder/test/unit/htmlcov' does not exist.
Recording test results
Sending e-mails to: chainsaw@redhat.com
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
Setting status of 725ad4c95d78e7c13eb0a1141b59ad56404dae22 to FAILURE with url http://rhsm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/python-rhsm-pull-request-builder/37/ and message: 'Build finished. 298 tests run, 0 skipped, 1 failed.'
```

**Comment by Ceiu** [comment]
ACK

**Pull request merged**